library/src/main/java/com/google/android/exoplayer/MediaCodecTrackRenderer.java

```diff
@@ -469,9 +488,10 @@ private void flushCodec() throws ExoPlaybackException {
     decodeOnlyPresentationTimestamps.clear();
     // Workaround for framework bugs.
     // See [Internal: b/8347958], [Internal: b/8578467], [Internal: b/8543366].
-    if (Util.SDK_INT >= 18) {
+    if (Util.SDK_INT >= 18 && codecReinitState == REINIT_STATE_NONE) {
```

Comment:
no because just calling flush() is not enough if we have a reinit pending due to an input format change, we really need to release and reinit the codec.